packages/react-components/src/components/TextFieldWithMention/TextFieldWithMention.tsx

```diff
@@ -0,0 +1,629 @@
+      textFieldRef?.current?.focus();
+      setActiveSuggestionIndex(undefined);
+      onChange && onChange(undefined, updatedContent.updatedHTML);
+    },
+    [
+      textFieldRef,
+      inputTextValue,
+      currentTriggerStartIndex,
+      mentionLookupOptions?.trigger,
+      onChange,
+      textValue,
+      tagsValue,
+      updateMentionSuggestions,
+      localeStrings
+    ]
+  );
+
+  const onTextFieldKeyDown = useCallback(
+    (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      // caretIndex should be set to undefined when the user is typing
+      setCaretIndex(undefined);
+      // shouldHandleOnMouseDownDuringSelect should be set to false after the last mouse down event.
+      // it shouldn't be updated in onMouseUp
+      // as onMouseUp can be triggered before or after onSelect event
+      // because its order depends on mouse events not selection.
+      setShouldHandleOnMouseDownDuringSelect(false);
+      if (isEnterKeyEventFromCompositionSession(ev)) {
+        return;
+      }
+      if (ev.key === 'ArrowUp') {
+        ev.preventDefault();
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? mentionSuggestions.length - 1
+              : Math.max(activeSuggestionIndex - 1, 0);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      } else if (ev.key === 'ArrowDown') {
+        ev.preventDefault();
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? 0
+              : Math.min(activeSuggestionIndex + 1, mentionSuggestions.length - 1);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      }
+      if (ev.key === 'Enter' && (ev.shiftKey === false || !supportNewline)) {
+        ev.preventDefault();
+        // If we are looking up a mention, select the focused suggestion
+        if (mentionSuggestions.length > 0 && activeSuggestionIndex !== undefined) {
+          const selectedMention = mentionSuggestions[activeSuggestionIndex];
+          if (selectedMention) {
+            onSuggestionSelected(selectedMention);
+            return;
+          }
+        }
+        onEnterKeyDown && onEnterKeyDown();
+      }
+      onKeyDown && onKeyDown(ev);
+    },
+    [onEnterKeyDown, onKeyDown, supportNewline, mentionSuggestions, activeSuggestionIndex, onSuggestionSelected]
+  );
+
+  const debouncedQueryUpdate = useDebouncedCallback(async (query?: string) => {
+    if (query === undefined) {
+      updateMentionSuggestions([]);
+      return;
+    }
+    const suggestions = (await mentionLookupOptions?.onQueryUpdated(query)) ?? [];
+    if (suggestions.length === 0) {
+      setActiveSuggestionIndex(undefined);
+    } else if (activeSuggestionIndex === undefined) {
+      setActiveSuggestionIndex(0);
+    }
+    updateMentionSuggestions(suggestions);
+  }, 500);
```

Comment:
Make this a constant like DEFAULT_MENTION_TRIGGER, put it at the very beginning of the file